Lib/textwrap.py

```diff
@@ -476,13 +476,16 @@ def indent(text, prefix, predicate=None):
     consist solely of whitespace characters.
     """
     if predicate is None:
-        def predicate(line):
-            return line.strip()
+        predicate = str.lstrip
 
-    def prefixed_lines():
-        for line in text.splitlines(True):
-            yield (prefix + line if predicate(line) else line)
-    return ''.join(prefixed_lines())
+    prefixed_lines = []
+    for line in text.splitlines(True):
+        if predicate(line):
+            prefixed_lines.extend((prefix, line))
```

Comment:
```suggestion
            prefixed_lines.append(prefix)
            prefixes_lines.append(line)
```
I do not have the possibility to benchmark this right now, but this would avoid creation of an intermediate tuple.

Reply:
Then, we can merge the else: clause.
I will do it.

Reply:
Does it really makes the code faster? Often the code uses `f.write(a+b)` instead of `f.write(a); f.write(b)`, because it is faster for small strings. It was an optimization technique in past, does it have an opposite effect now?

Reply:
> Does it really makes the code faster? Often the code uses `f.write(a+b)` instead of `f.write(a); f.write(b)`, because it is faster for small strings. It was an optimization technique in past, does it have an opposite effect now?

Yes. list.append(prefix+line) = 2.32ms in the benchmark.

Writing to file and list.append is different.

* file.write needs thread lock.
* file.write(a); file.write(b); needs a+b space always. `list.extend((a,b))` can share `a` if it appeares in the list many times. (better cache localicity).

Reply:
```py
if predicate(line):
    prefixed_lines.append(prefix)
prefixes_lines.append(line)
```

1.81ms!